### PR TITLE
rpc: Add testblocktemplatevalidity

### PR DESCRIPTION
**Pull request by maflcko**
A caller can check if a block template is valid and is good to spend work on

### CODE REVIEW

**Comment by MarcoFalke** [comment]
TODO: Add tests (Currently left out to avoid excessive merge conflicts with other pull requests touching the miner tests)

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -258,6 +258,50 @@ static UniValue prioritisetransaction(const JSONRPCRequest& request)
...
+    if (block.hashPrevBlock != tip->GetBlockHash()) {
+        throw JSONRPCError(RPC_VERIFY_ERROR, "Must submit block that is built on the current tip (" + tip->GetBlockHash().GetHex() + ")");
+    }
+
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("block_hash", block.GetHash().GetHex());
+    CValidationState state;
+    bool valid_template = TestBlockValidity(state, Params(), block, /* pindexPrev */ tip, /* fCheckPOW */ false, /* fCheckMerkleRoot */ true);
+    result.pushKV("valid_template", valid_template);
+    if (valid_template) {
```
Comment (Empact) [inline src/rpc/mining.cpp]: Is this inverted?

**Comment by luke-jr** [comment]
This is already supported via BIP 23 Block Proposals: https://github.com/bitcoin/bips/blob/master/bip-0023.mediawiki#Block_Proposal

**Comment by promag** [comment]
Right, looks like this is already implemented
https://github.com/bitcoin/bitcoin/blob/56f69360dc98bd68704f19646a84d045788d199e/src/rpc/mining.cpp#L389-L416

**Pull request closed without merging**